api/src/shared/spec/security.ts

```diff
@@ -0,0 +1,26 @@
+import assert from 'assert';
+
+import status from 'http-status-codes';
+
+import testApi from './api';
+
+export function createPostTestForGuardAgainstUnlogin(
+  url: string,
+  payload: string | object
+) {
+  return async () => {
+    const resp = await testApi('anyInvalidToken')
+      .post(url)
+      .send(payload);
+
+    assert.strictEqual(resp.status, status.UNAUTHORIZED);
+  };
+}
+
+export function createGetTestForGuardAgainstUnlogin(url: string) {
+  return async () => {
+    const resp = await testApi('anyInvalidToken').get(url);
+
+    assert.strictEqual(resp.status, status.UNAUTHORIZED);
+  };
+}
```

Comment:
this is strange to create such test for every method and you potentially can need many of this methods, e.g. for specific headers, query, etc.

my proposition is to create an api smth like that:
```ts
// this need better typing, is just for showing the idea
export function shouldBlockUnauthorizedUser(callback: (api: Api) => Promise) {
   const res = await callback(testApi('wrong-token'));
   assert.strictEqual(resp.status, status.UNAUTHORIZED);
}

// usage:
shouldBlockUnauthizedUser(api => api.post(url).send(payload));
```
by this you can handle every possible call with single, simple function